File: observability-service/internal/core/domain/insights.go
package domain

import "time"

type InsightsResponse struct {
	Recommendations []InsightRecommendation `json:"recommendations"`
	Anomalies       []InsightAnomaly        `json:"anomalies"`
	CostSavings     CostSavingsInsight      `json:"costSavings"`
	SLOStatus       SLOStatus               `json:"sloStatus"`
	Summary         InsightsSummary         `json:"summary"`
	Metadata        InsightsMetadata        `json:"metadata"`
	GeneratedAt     time.Time               `json:"generatedAt"`
}

type AIInsightsResponse struct {
	Recommendations []InsightRecommendation `json:"recommendations"`
	SummaryNote     string                  `json:"summaryNote,omitempty"`
}

type InsightRecommendation struct {
	Type           string  `json:"type"`
	Severity       string  `json:"severity"`
	Resource       string  `json:"resource"`
	Current        string  `json:"current"`
	Recommendation string  `json:"recommendation"`
	Savings        string  `json:"savings,omitempty"`
	Impact         string  `json:"impact,omitempty"`
	Confidence     float64 `json:"confidence,omitempty"`
	BasedOn        string  `json:"basedOn,omitempty"`
}

type InsightAnomaly struct {
	DetectedAt time.Time `json:"detectedAt"`
	Type       string    `json:"type"`
	Resource   string    `json:"resource"`
	Value      string    `json:"value"`
	Baseline   string    `json:"baseline"`
	Current    string    `json:"current"`
	Severity   string    `json:"severity"`
}

type CostSavingsInsight struct {
	IdleCPUCores        float64 `json:"idleCpuCores"`
	IdleMemoryGB        float64 `json:"idleMemoryGb"`
	EstimatedMonthlyUSD float64 `json:"estimatedMonthlyUsd"`
	Note                string  `json:"note,omitempty"`
}

type SLOStatus struct {
	SLOTargetPercent            float64 `json:"sloTargetPercent"`
	ErrorRatePercent            float64 `json:"errorRatePercent"`
	WithinSLO                   bool    `json:"withinSLO"`
	ErrorBudgetRemainingPercent float64 `json:"errorBudgetRemainingPercent"`
	Note                        string  `json:"note,omitempty"`
}

type InsightsSummary struct {
	TotalNamespaces      int  `json:"totalNamespaces"`
	TotalRecommendations int  `json:"totalRecommendations"`
	TotalAnomalies       int  `json:"totalAnomalies"`
	DataPointsProcessed  int  `json:"dataPointsProcessed"`
	PartialData          bool `json:"partialData"`
}

type InsightsMetadata struct {
	GenerationMs   int64                     `json:"generationMs"`
	CacheHit       bool                      `json:"cacheHit"`
	AnalysisWindow string                    `json:"analysisWindow"`
	ThresholdsUsed InsightsThresholdMetadata `json:"thresholdsUsed"`
}

type InsightsThresholdMetadata struct {
	AnomalyStdDev             float64 `json:"anomalyStdDev"`
	SLOErrorRateTargetPercent float64 `json:"sloErrorRateTargetPercent"`
}

File: frontend/src/features/admin/types/insights.ts
export type InsightSeverity = "low" | "medium" | "high" | "critical";
export type InsightType =
  | "right_sizing"
  | "scaling"
  | "reliability"
  | "anomaly"
  | "cost_optimization"
  | "slo";

export interface InsightRecommendation {
  type: string;
  severity: InsightSeverity;
  resource: string;
  current: string;
  recommendation: string;
  savings?: string;
  impact?: string;
  confidence?: number;
  basedOn?: string;
}

export interface InsightAnomaly {
  detectedAt: string;
  type: string;
  resource: string;
  value: string;
  baseline: string;
  current: string;
  severity: InsightSeverity;
}

export interface CostSavingsInsight {
  idleCpuCores: number;
  idleMemoryGb: number;
  estimatedMonthlyUsd: number;
  note?: string;
}

export interface SLOStatus {
  sloTargetPercent: number;
  errorRatePercent: number;
  withinSLO: boolean;
  errorBudgetRemainingPercent: number;
  note?: string;
}

export interface InsightsSummary {
  totalNamespaces: number;
  totalRecommendations: number;
  totalAnomalies: number;
  dataPointsProcessed: number;
  partialData: boolean;
}

export interface InsightsMetadata {
  generationMs: number;
  cacheHit: boolean;
  analysisWindow: string;
  thresholdsUsed: {
    anomalyStdDev: number;
    sloErrorRateTargetPercent: number;
  };
}

export interface InsightsResponse {
  recommendations: InsightRecommendation[];
  anomalies: InsightAnomaly[];
  costSavings: CostSavingsInsight;
  sloStatus: SLOStatus;
  summary: InsightsSummary;
  metadata: InsightsMetadata;
  generatedAt: string;
}

export const SEVERITY_CLASSES: Record<InsightSeverity, string> = {
  low: "bg-blue-500/15 text-blue-300 border-blue-500/30",
  medium: "bg-yellow-500/15 text-yellow-300 border-yellow-500/30",
  high: "bg-orange-500/15 text-orange-300 border-orange-500/30",
  critical: "bg-red-500/15 text-red-300 border-red-500/30"
};
